docs/website/docs/tutorial/filesystem.md

```diff
@@ -0,0 +1,355 @@
+
+1. Accessing files in the bucket / directory.
+2. Reading the files and yielding records.
+
+`dlt`'s filesystem source includes several resources that you can use together or individually:
+
+- `filesystem` resource accesses files in the directory or bucket
+- several readers resources (`read_csv`, `read_parquet`, `read_jsonl`) read files and yield the records
+
+Let's initialize a source and create a pipeline for loading `csv` files from Google Cloud Storage to DuckDB. You can replace code from `filesystem_pipeline.py` with the following:
+
+```py
+import dlt
+from dlt.sources.filesystem import filesystem, read_csv
+
+files = filesystem(bucket_url="gs://filesystem-tutorial", file_glob="encounters*.csv")
+reader = (files | read_csv()).with_name("encounters")
+pipeline = dlt.pipeline(pipeline_name="hospital_data_pipeline", dataset_name="hospital_data", destination="duckdb")
+info = pipeline.run(reader)
+print(info)
+```
+
+What's happening in this snippet?
+
+1. We initialize the filesystem resource and pass the `file_glob` parameter. Based on this parameter, `dlt` will filter all files in the bucket.
+2. We pipe the transformer `read_csv` to read the files yielded by the filesystem resource and iterate over records from the file.
+
+:::note
+A **transformer** in `dlt` is a special type of resource that processes each record from another resource. This lets you chain multiple resources together. To learn more, check out the [transformer section](../general-usage/resource#process-resources-with-dlttransformer).
+:::
+
+3. We create the `dlt` pipeline with the name `hospital_data_pipeline` and DuckDB destination and run this pipeline.
+
+## 3. Configuring filesystem source
+
+:::note
+In this tutorial we will work with publicly accessed dataset [Hospital Patient Records](https://mavenanalytics.io/data-playground?order=date_added%2Cdesc&search=Hospital%20Patient%20Records)
+synthetic electronic health care records. You can use the exact credentials from this tutorial to load this dataset from GCP.
+<details>
+<summary>Citation</summary>
+Jason Walonoski, Mark Kramer, Joseph Nichols, Andre Quina, Chris Moesel, Dylan Hall, Carlton Duffett, Kudakwashe Dube, Thomas Gallagher, Scott McLachlan, Synthea: An approach, method, and software mechanism for generating synthetic patients and the synthetic electronic health care record, Journal of the American Medical Informatics Association, Volume 25, Issue 3, March 2018, Pages 230–238, https://doi.org/10.1093/jamia/ocx079
+</details>
+:::
+
+Next, we need to configure the connection. Specifically, we’ll set the bucket URL and credentials. This example focuses on Google Cloud Storage. For other Cloud Storage services, see the [Filesystem configuration section](../dlt-ecosystem/verified-sources/filesystem/basic#configuration).
+
+Let's specify the bucket URL and credentials:
+
+<Tabs
```

Comment:
I think for the tutorial we can just pick one type of configuration (the most common/convenient for the tutorial) and just link the docs for alternative ways.

Reply:
I thought this two at first, but one of the users complained to me that we have examples with toml files everywhere and it's difficult to get other examples. I think it doesn't hurt to have options, wdyt?